tests/flex/test_pythonfunctioncontainer.py

```diff
@@ -61,3 +66,13 @@ def test_with_executor_wait(self):
             self.assertFalse(job.server.future.done())
             self.project.wait_for_job(job=job, interval_in_s=0.01, max_iterations=1000)
             self.assertTrue(job.server.future.done())
+
+    def test_with_internal_executor(self):
+        job = self.project.wrap_python_function(my_function_exe)
+        job.input["a_lst"] = [1, 2, 3, 4]
+        job.input["b_lst"] = [5, 6, 7, 8]
+        job.server.cores = 2
+        job.executor_type = "ProcessPoolExecutor"
```

Comment:
Tests don't handle failure case; what happens if I assign something weird here? What should happen/when should it happen? IMO behaviour should be to get an exception at assignment time, and that should be codified in the tests, but I think right now you don't hit problems until you call `run` and I don't know whether the error message is informative or not.

Reply:
I added additional tests for the failure case.